### PR TITLE
Add checks to ensure only 'default' can be used as name of task platform

### LINKED ISSUE #2710: Add checks for configuration of task platform other than `default`

### PR DESCRIPTION
**Pull request by markpollack**
Fixes #2710

### CODE REVIEW

**Comment by chrisjs** [comment]
looks good to merge but travis is failing

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/AboutController.java [hunk 1]
```diff
@@ -23,6 +23,7 @@
 
 import org.springframework.beans.factory.annotation.Value;
 import org.springframework.cloud.common.security.support.SecurityStateBean;
+import org.springframework.cloud.dataflow.core.Launcher;
```
Comment (cppwfs) [inline spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/AboutController.java]: checkstyle is not happy.  Can fix on merge.

**Comment by cppwfs** [comment]
LGTM.   Will fix the checkstyle that caused the failure.

**Comment by cppwfs** [comment]
FIxed checkstyle issue, rebased, squashed merged.

**Pull request closed without merging**